docs/src/index.md

```diff
@@ -315,3 +315,55 @@ To get back an `Array` from a `DArray`, just call `collect`:
 DA = rand(Blocks(32, 32), 256, 128)
 collect(DA) # returns a `Matrix{Float64}`
 ```
+
+## Quickstart: Datadeps
+
+Datadeps is a feature in Dagger.jl that facilitates parallelism control within designated regions, allowing tasks to write to their arguments while ensuring dependencies are respected.
+For more details: [Datadeps](@ref)
+
+### Syntax
+The Dagger.spawn_datadeps() function is used to create a "datadeps region" where tasks are executed with parallelism controlled by specified dependencies.
+
+```julia
+Dagger.spawn_datadeps() do
+    Dagger.@spawn func!(Out(var_name_x), In(var_name_y))
+end
+```
+### Argument Annotation
+
+- In(X): Indicates that the variable X is an input to the task.
+- Out(X): Indicates that the variable X is an output of the task.
+- InOut: Indicates that the variable is both an input and an output.
+
+### Example with Datadeps
+
+```julia
+X = [4,5,6,7,1,2,3,9,8]
+C = zeros(10)
+
+Dagger.spawn_datadeps() do
+    Dagger.@spawn sort!(InOut(X))
+    Dagger.@spawn copyto!(Out(C), In(X))
+end
+
+# C = [1,2,3,4,5,6,7,8,9]
+```
+
+In this example, the sort! function operates on array X, while the copyto! task operates on array C. By specifying dependencies using argument annotations, the tasks are executed in a controlled parallel manner, resulting in a sorted C array.
+
+### Example without Datadeps
+
+```julia
+X = [4,5,6,7,1,2,3,9,8]
+C = zeros(10)
+Dagger.spawn_datadeps() do
+    Dagger.@spawn sort!(X)
+    Dagger.@spawn copyto!(C, X)
+end
```

Comment:
```suggestion
Dagger.@spawn sort!(X)
Dagger.@spawn copyto!(C, X)
```

Technically it's functionally identical, but I think it's a bit confusing if we say "without Datadeps", but then still have the `spawn_datadeps` region.

Reply:
Alternatively, you could leave `spawn_datadeps` here, but change the header to "Example with Datadeps (without annotations)" or something like that.